### PR TITLE
Logging guidelines draft

### PR DESCRIPTION
**Pull request by stranger80**

### CODE REVIEW

**Review by shadeofblue — APPROVED** [review]
nicely summarized, nice 👍

File: docs/logging-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+## Log level guidelines
+
+### CRITICAL/FATAL
+**Purpose:** 
+- Indicate the app is unable to continue, it should exit after this message.
+
+**Audience:**
+- Users
+
```
Comment (filipgolem) [inline docs/logging-guidelines.md:77]: I think this is also very important for node owners/admins.

File: docs/logging-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+As low-level, 3rd party library errors are encountered during execution, their error messages are usually useless without context information. It is vital to wrap the low-level error messages with additional info to indicate details of performed activity that would aid in troubleshooting. Eg. instead of:
+
+```
+[2020-08-27T07:56:22Z ERROR yagna] File IO error: Path not found
+```
+
+log this:
+
+```
+[2020-08-27T07:56:22Z ERROR yagna] E00342 - WASM ExeUnit DEPLOY: Error downloading remote file to temp folder '/tmp/yagna_data': File IO error: Path not found
```
Comment (filipgolem) [inline docs/logging-guidelines.md:165]: From my experience of debugging yagna: the file name and line number where the error occurred is very important as it's ofter easier to just read the source code to find the problem. (I know you mentioned it in "Log entry content" section.)

**Review by filipgolem — DISMISSED** [review]
Very good document.

File: docs/logging-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+The log entries should record following aspects/attributes:
+
+**Must have:**
+- **Timestamp** (with ms granularity, in UTC or with TZ information) - must be possible to correlate entries from nodes in different timezones
+- Log **level**
+- Log **area/topic** (eg. namespace or module of code which records the log entry)
+- **Grouping or correlating** attribute (attribute which allows to filter events related to a single command, activity, API request, etc.)
+  - Thread id
+  - Bespoke correlation id
+- Log entry **description** (human readable, no linebreaks)
```
Comment (nieznanysprawiciel) [inline docs/logging-guidelines.md:44]: Sometimes "no linebreaks" is opposite of "human readable"

**Comment by stranger80** [comment]
Merged latest master state and added reference to logging guidelines from the SDP doc.

File: docs/logging-guidelines.md (hunk 2 above)
Comment (stranger80) [reply docs/logging-guidelines.md:165]: Agreed, albeit the error code indicated in this example serves this purpose: with a specific error code it should be fairly instantaneous to identify the location responsible for a particular issue (I'm conscious that providing file/line number may not be feasible for many dev platforms).

File: docs/logging-guidelines.md (hunk 3 above)
Comment (stranger80) [reply docs/logging-guidelines.md:44]: Agreed, rewritten

**Review by shadeofblue — APPROVED** [review]

File: docs/development-plan.md [hunk 4]
```diff
@@ -106,6 +106,10 @@ https://doc.rust-lang.org/1.0.0/style/README.html
 To enforce formatting, code should be formatted using rustfmt tool (https://github.com/rust-lang/rustfmt).
 To install it, run `rustup component add rustfmt` command. To format files in the working dictory, please run `cargo fmt` command.
 
+### Logging Guidelines
+
+The logging guidelines for Yagna component development can be found here: logging-guidelines.md.
```
Comment (tworec) [inline docs/development-plan.md:111]: this does not render as link

**Comment by tworec** [comment]
👍 good job guys! Thx!

**Pull request merged**